From 84041d73eec4e62b6ee49a467dd455f7454c78ae Mon Sep 17 00:00:00 2001
From: Andrew Jeffery <andrew@aj.id.au>
Date: Fri, 17 Nov 2017 12:19:29 +1030
Subject: [PATCH 2/3] automake: Make automake.in compatible with Perl 5.22

Left-brace in regex needs to be escaped as of 5.22.

Change-Id: Ie432453783dcee4d9fe49eadb0bb70e6ca06f0c3
Signed-off-by: Andrew Jeffery <andrew@aj.id.au>
---
 ...n-Left-brace-in-regex-needs-to-be-escaped.patch | 31 ++++++++++++++++++++++
 .../recipes-devtools/automake/automake_1.15.bb     |  3 ++-
 2 files changed, 33 insertions(+), 1 deletion(-)
 create mode 100644 import-layers/yocto-poky/meta/recipes-devtools/automake/automake/0001-automake.in-Left-brace-in-regex-needs-to-be-escaped.patch

diff --git a/import-layers/yocto-poky/meta/recipes-devtools/automake/automake/0001-automake.in-Left-brace-in-regex-needs-to-be-escaped.patch b/import-layers/yocto-poky/meta/recipes-devtools/automake/automake/0001-automake.in-Left-brace-in-regex-needs-to-be-escaped.patch
new file mode 100644
index 000000000000..733aa517cad0
--- /dev/null
+++ b/import-layers/yocto-poky/meta/recipes-devtools/automake/automake/0001-automake.in-Left-brace-in-regex-needs-to-be-escaped.patch
@@ -0,0 +1,31 @@
+From eb0bb2ce7dec1856c78ee2f3df44db9749e0c024 Mon Sep 17 00:00:00 2001
+From: Andrew Jeffery <andrew@aj.id.au>
+Date: Fri, 17 Nov 2017 12:16:15 +1030
+Subject: [PATCH] automake.in: Left-brace in regex needs to be escaped
+
+Announced in Perl 5.16, enforced as of 5.22[1]. Use the escape to avoid errors
+when building with modern perl interpreters.
+
+[1] http://search.cpan.org/dist/perl-5.22.0/pod/perldelta.pod#A_literal_%22%7B%22_should_now_be_escaped_in_a_pattern
+
+Signed-off-by: Andrew Jeffery <andrew@aj.id.au>
+---
+ bin/automake.in | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/bin/automake.in b/bin/automake.in
+index eedc8bcf196c..a679d1657bbd 100644
+--- a/bin/automake.in
++++ b/bin/automake.in
+@@ -3878,7 +3878,7 @@ sub substitute_ac_subst_variables_worker
+ sub substitute_ac_subst_variables
+ {
+   my ($text) = @_;
+-  $text =~ s/\${([^ \t=:+{}]+)}/substitute_ac_subst_variables_worker ($1)/ge;
++  $text =~ s/\$\{([^ \t=:+{}]+)}/substitute_ac_subst_variables_worker ($1)/ge;
+   return $text;
+ }
+ 
+-- 
+2.14.1
+
diff --git a/import-layers/yocto-poky/meta/recipes-devtools/automake/automake_1.15.bb b/import-layers/yocto-poky/meta/recipes-devtools/automake/automake_1.15.bb
index 7b33b9a33c40..385e3f2247a9 100644
--- a/import-layers/yocto-poky/meta/recipes-devtools/automake/automake_1.15.bb
+++ b/import-layers/yocto-poky/meta/recipes-devtools/automake/automake_1.15.bb
@@ -22,7 +22,8 @@ SRC_URI += " file://python-libdir.patch \
             file://buildtest.patch \
             file://performance.patch \
             file://new_rt_path_for_test-driver.patch \
-            file://0001-doc-Don-t-discard-stderr-when-invoking-help2man.patch"
+            file://0001-doc-Don-t-discard-stderr-when-invoking-help2man.patch \
+            file://0001-automake.in-Left-brace-in-regex-needs-to-be-escaped.patch"
 
 SRC_URI[md5sum] = "716946a105ca228ab545fc37a70df3a3"
 SRC_URI[sha256sum] = "7946e945a96e28152ba5a6beb0625ca715c6e32ac55f2e353ef54def0c8ed924"
-- 
2.14.1

